### PR TITLE
[dev/eng] Add testing commands to netci.groovy

### PR DESCRIPTION
**Pull request by mellinoe**
This turns on testing in Jenkins for dev/eng.

@chcosta @karajas @weshaggard

### CODE REVIEW

**Comment by mellinoe** [comment]
@dotnet-bot Test CI Please

**Comment by mellinoe** [comment]
https://ci.dot.net/job/dotnet_corefx/job/dev_eng/job/GenPRTest/job/windows_nt_debug_prtest/1/console
https://ci.dot.net/job/dotnet_corefx/job/dev_eng/job/GenPRTest/job/osx_debug_prtest/1/console
https://ci.dot.net/job/dotnet_corefx/job/dev_eng/job/GenPRTest/job/ubuntu16.04_debug_prtest/

**Comment by mellinoe** [comment]
I needed to cherry-pick a commit for some extra test failures on win7 / x86 (never tested that locally. We have a green build on Windows, Ubuntu, and OSX, so this seems good to go in to me.

File: netci.groovy [hunk 1]
```diff
@@ -230,13 +230,16 @@ def testNugetRuntimeIdConfiguration = ['Debug': 'win7-x86',
...
                     }
                     else if (osName == 'OSX') {
-                        shell("HOME=\$WORKSPACE/tempHome ./build.sh -${configurationGroup.toLowerCase()} -outerloop -- /p:TestWithLocalNativeLibraries=true /p:WithoutCategories=IgnoreForCI")
+                        shell("HOME=\$WORKSPACE/tempHome ./build.sh -${configurationGroup.toLowerCase()}")
+                        shell("HOME=\$WORKSPACE/tempHome ./build-tests.sh -${configurationGroup.toLowerCase()} -outerloop -- /p:WithoutCategories=IgnoreForCI")
                     }
                     else {
-                        shell("sudo HOME=\$WORKSPACE/tempHome ./build.sh -${configurationGroup.toLowerCase()} -outerloop -- /p:TestWithLocalNativeLibraries=true /p:TestNugetRuntimeId=${targetNugetRuntimeMap[osName]} /p:WithoutCategories=IgnoreForCI")
+                        shell("sudo HOME=\$WORKSPACE/tempHome ./build.sh -${configurationGroup.toLowerCase()}")
+                        shell("sudo HOME=\$WORKSPACE/tempHome ./build-tests.sh -${configurationGroup.toLowerCase()} -outerloop -- /p:TestNugetRuntimeId=${targetNugetRuntimeMap[osName]} /p:WithoutCategories=IgnoreForCI")
```
Comment (weshaggard) [inline netci.groovy]: Can you remove the TestNugetRuntimeId as it should no longer be necessary.

File: netci.groovy [hunk 2]
```diff
@@ -316,13 +319,15 @@ def testNugetRuntimeIdConfiguration = ['Debug': 'win7-x86',
...
+                            batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && build.cmd -${configurationGroup} -os:${osGroup} -buildArch:${buildArchConfiguration[configurationGroup]} -- /p:BuildConfiguration=${targetGroup}-${osGroup}-${configurationGroup}-${buildArchConfiguration[configurationGroup]}")
+                            batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && build-tests.cmd -${configurationGroup} -os:${osGroup} -buildArch:${buildArchConfiguration[configurationGroup]} -TestNugetRuntimeId:${testNugetRuntimeIdConfiguration[configurationGroup]} -- /p:WithoutCategories=IgnoreForCI /p:BuildConfiguration=${targetGroup}-${osGroup}-${configurationGroup}-${buildArchConfiguration[configurationGroup]}")
```
Comment (weshaggard) [inline netci.groovy]: Remove TestNugetRuntimeId.

**Comment by mellinoe** [comment]
I'll merge when this succeeds: https://ci.dot.net/job/dotnet_corefx/job/dev_eng/job/GenPRTest/job/windows_nt_debug_prtest/4/

**Comment by mellinoe** [comment]
@dotnet-bot Test CI please

**Comment by mellinoe** [comment]
https://ci.dot.net/job/dotnet_corefx/job/dev_eng/job/GenPRTest/job/windows_nt_debug_prtest/5/console

**Pull request merged**